eng/testing/tests.mobile.targets

```diff
@@ -157,16 +157,16 @@
       </WasmSatelliteAssemblies>
     </ItemGroup>
     <ItemGroup>
-      <AssemblySearchPaths Include="$(PublishDir)"/>
+      <Assemblies Include="$(PublishDir)\*.dll"/>
       <WasmFilesToIncludeInFileSystem Include="@(ContentWithTargetPath)" />
       <WasmFilesToIncludeInFileSystem Include="@(ReferenceCopyLocalPaths)" Condition="'%(ReferenceCopyLocalPaths.BuildReference)' == 'true' and !$([System.String]::new('%(ReferenceCopyLocalPaths.Identity)').EndsWith('.resources.dll'))" />
       <WasmFilesToIncludeInFileSystem Include="@(WasmSatelliteAssemblies)" TargetPath="%(WasmSatelliteAssemblies.CultureName)\%(WasmSatelliteAssemblies.Filename)%(WasmSatelliteAssemblies.Extension)" />
       <!-- Include files specified by test projects from publish dir -->
       <WasmFilesToIncludeInFileSystem Include="@(WasmFilesToIncludeFromPublishDir -> '$(PublishDir)%(Identity)')" />
-      <ExtraAssemblies Include="$(PublishDir)$(AssemblyName).dll" />
+      <Assemblies Include="$(PublishDir)$(AssemblyName).dll" />
       <!-- We need these facades for some tests. -->
-      <ExtraAssemblies Include="$(PublishDir)mscorlib.dll" />
-      <ExtraAssemblies Include="$(PublishDir)System.Drawing.dll" />
+      <Assemblies Include="$(PublishDir)mscorlib.dll" />
+      <Assemblies Include="$(PublishDir)System.Drawing.dll" />
     </ItemGroup>
     <Error Condition="!Exists('$(MicrosoftNetCoreAppRuntimePackRidDir)')" Text="MicrosoftNetCoreAppRuntimePackRidDir=$(MicrosoftNetCoreAppRuntimePackRidDir) doesn't exist" />
     <WasmAppBuilder
```

Comment:
Why is the new task not being used with this? IIUC, that would be needed to retain the existing "behavior" at least, and then a follow up PR can make the changes for streamlining that, or removing MLC etc.

Reply:
And with that, you shouldn't need the change on line 160, or adding the `Import`

Reply:
The reason why the new task exists is to make sure the runtime tests can still simulate a publish.  It's not needed anywhere else.  The libraries tests already publish.

Reply:
I'm trying to understand why this is needed in here:
```diff
-      <AssemblySearchPaths Include="$(PublishDir)"/>
+       <Assemblies Include="$(PublishDir)\*.dll"/>
```

.. and
```diff
+ <Import Project="$(RepositoryEngineeringDir)illink.targets" Condition="'$(TargetOS)' == 'Browser'" />
```

Reply:
We need to just carry over whatever is in the publish dir assemblies wise, so `<Assemblies Include="$(PublishDir)\*.dll"/>` is probably the only thing we need.  My guess about the others is that previously we were still going through MDLC and needed to force mscorlib, etc along.  

No idea about the linker import.  @mdh1418 ?

Reply:
The import overrides the current linker being used so that it will use the linker from a nuget package rather than the one shipped from the dotnet sdk. The reason for this is that there was an error being thrown along the lines of `System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')`. The was already a fix for it but it's more recent and didn't yet make it to the current sdk being used. It was induced by adding `<PublishTrimmed>true</PublishTrimmed>` revealing the error, but we should still use it (regardless/independent from `<PublishTrimmed>`).

The original code will have resolved `MainAssembly` using paths provided in `AssemblySearchPaths` to build the WasmApp with necessary assemblies. However, these assemblies are now already resolved and placed in the `$(PublishDir)` as a result of publishing (I imagine it wasn't the case when `WasmAppBuilder` was first written). So now we can just include the assemblies necessary to run the app solely within the `Assemblies` parameter rather than needing to resolve them again (which needs to be done for the debugger-tests and runtime tests since they do not publish).

Reply:
>The original code will have resolved MainAssembly using paths provided in AssemblySearchPaths to build the WasmApp with necessary assemblies. However, these assemblies are now already resolved and placed in the $(PublishDir) as a result of publishing (I imagine it wasn't the case when WasmAppBuilder was first written). So now we can just include the assemblies necessary to run the app solely within the Assemblies parameter rather than needing to resolve them again (which needs to be done for the debugger-tests and runtime tests since they do not publish).

Great! Can you include this in final commit message/PR description too?

> The was already a fix for it but it's more recent and didn't yet make it to the current sdk being used. It was induced by adding <PublishTrimmed>true</PublishTrimmed> revealing the error, but we should still use it (regardless/independent from <PublishTrimmed>).

I'm not sure I understand why it needs to be added here. IIUC, it is not being enabled here, and `illink.targets` should not be needed here.

Reply:
Done, modified a bit so it would be a bit less wordy :)